core/insertion_marker_manager.ts

```diff
@@ -658,39 +630,39 @@ export class InsertionMarkerManager {
       imConn.connect(closest);
     }
 
-    this.markerConnection_ = imConn;
+    this.markerConnection = imConn;
   }
 
   /**
    * Disconnects and hides the current insertion marker. Should return the
    * blocks to their original state.
+   *
+   * @param markerConnection Connection on the insertion marker block that
+   *     corresponds to the active candidate's local connection on the
+   *     currently dragged block.
    */
-  private hideInsertionMarker_() {
-    if (!this.markerConnection_) {
-      console.log('No insertion marker connection to disconnect');
-      return;
-    }
-
-    const imConn = this.markerConnection_;
-    const imBlock = imConn.getSourceBlock();
+  private hideInsertionMarker(markerConnection: RenderedConnection) {
+    const imBlock = markerConnection.getSourceBlock();
```

Comment:
Why not just make this non-throwing, and then call it unconditionally?:
```suggestion
  private hideInsertionMarker() {
    if (!this.markerConnection) return;
    const imBlock = this.markerConnection.getSourceBlock();
```
…and likewise for `hideInsertionInputOutline` and `hideReplacementFade`.